src/iop/demosaicing/rcd.c

```diff
@@ -189,7 +189,8 @@ static void rcd_ppg_border(
         color[1] = pc;
 
       color[3] = 0.0f;
-      memcpy(buf, color, sizeof(float) * 4);
+      for_each_channel(k)
+        buf[k] = color[k];
```

Comment:
Not for this PR, but if `buf` is aligned, this can be replaced by `copy_pixel(buf, color)`, at which point it's worth checking whether `copy_pixel_nontemporal` gives a performance boost at high thread counts.

Reply:
I have looked at code where you used it.

There is one general question involved, years back i did a lot of real-signal processing and i found it very difficult to benchmark in a way, that is valid for many / different architectures. Cache to memory writes depend so much on how the cpu connects to ram. Also not all cpu's handle writing "partially" to cachelines well.

I would assume on cpu's with a big level 3 cache it would be likely we have valid data for the next module in the pixelpipe still in cache, maybe less likely while exporting but very much while in darkroom using roi.

Any ideas on this?

Reply:
Nontemporal writes avoid reading the cache line into cache and use a separate set of write buffers which don't have to load the cache line before you can modify any of the bytes in the line.
Even with 128MB L3 cache and quad-channel memory, I was seeing 30% reduction in runtime on the simpler IOPs which only read the source once and write the destination once - from the reduction in memory bandwidth by eliminating the second read to load the destination's cache line.